index.bs

```diff
@@ -4684,26 +4702,47 @@ To <dfn>parse URL pattern</dfn>, given |pattern|:
      <dd>|url|'s [=url/host=] if |has hostname| is true, or null otherwise.
      <dt>port
      <dd>
-     1. If |url|'s [=url/scheme=] [=is special=] and |url|'s
-   [=url/scheme=]'s [=default port=] is not null, and |url|'s [=url/port=] is
-   null or is equal to [=url/scheme=]'s [=default port=]:
+     1. If |has port| is false:
 
-       1. The empty string.
+       1. null.
 
-     1. Otherwise, if |url|'s [=url/port=] is not null:
+     1. Otherwise:
 
-       1. [=Serialize an integer=] with |url|'s [=url/port=].
+       1. If |url|'s [=url/scheme=] [=is special=] and |url|'s [=url/scheme=]'s
+          [=default port=] is not null, and |url|'s [=url/port=] is null or is
+          equal to [=url/scheme=]'s [=default port=]:
 
-     1. Otherwise:
+         1. The empty string.
 
-       1. null.
+       1. Otherwise, if |url|'s [=url/port=] is not null:
+
+         1. [=Serialize an integer=] with |url|'s [=url/port=].
```

Comment:
Do you also need to mention here the integer range for a valid port number?

Reply:
At the moment we don't have to: this will already raise an error when trying to parse the URL, near the end of the parsing step. 

Now if we stop using the basic URL parser here (and it might make sense based on https://github.com/w3c/webdriver-bidi/issues/507#issuecomment-1688462008), adding this validation would make sense.